docs/troubleshoot/app-framework/app-troubleshoot.md

```diff
@@ -223,18 +223,38 @@ Add the Zowe Desktop directory path to the `MVD_DESKTOP_DIR` environment variabl
   set MVD_DESKTOP_DIR=<zlux-root-dir>/zlux-app-manager/virtual-desktop
   ```
 
+
 ## Error: Exception thrown when reading SAF keyring {ZWED0148E}
 
 **Symptom:**
-The error message indicates that Zowe's local certificate authority (local CA) `ZoweCert`, the certificate `jwtsecret`, or the Zowe certificate `localhost` does not exist in the Zowe keyring. ZWED0148E contains the following messages.
+
+If you see one or more of the following in the logs,
+
+
+- ZWED0148E - Exception thrown when reading SAF keyring, e= Error: R_datalib call failed: function code: 01, SAF rc: `number`, RACF rc: `number`, RACF rsn: `number`
+
+
+* java.io.IOException: R_datalib (IRRSDL00) error: profile for ring not found (`number`, `number`, `number`)
+
+
+Then the cause is due to keyring configuration.
+You may also see the log message
+
+`ZWES1060W Failed to init TLS environment, rc=1(Handle is not valid)`
+
+ **Note:** This log message can have other causes too, such as lack of READ permission to resources in the CRYPTOZ class.
+
+**Solution:**
+
+Refer to table 2 (DataGetFirst) of the [Return and Reason Codes](https://www.ibm.com/docs/en/zos/2.5.0?topic=library-return-reason-codes) to decide the problem you are facing.

```

Comment:
The table name is capitalized, so it should be capitalized here too -- and, if I were writing this, I'd include the full table name, but that's up to you: Refer to Table 2 (DataGetFirst and DataGetNext return and reason codes)

Also, the reader is not going to "decide" the problem they're having. The reader can try to figure it out:  "Refer to Table 2 (...) of Return and Reason Codes to try to find/identify/figure out the problem"